lib/elixir/lib/module/types/pattern.ex

```diff
@@ -121,6 +121,7 @@ defmodule Module.Types.Pattern do
     {:ok, :dynamic, context}
   end
 
+  # TODO: What about maps dude?
```

Comment:
As I mentioned above, I don't think you need to do anything here. For the checker, `typeof(^expr)` will always be the same as `typeof(expr)`. :) So you are off the hook!

Reply:
Sorry, accidentally left this comment and the above in. I will clarify or remove them.

Reply:
For example, adding a test for `a = :a; %{^a => :ok} = %{:b => :error}` should be enough to test this works as expected.

Reply:
Map checks in patterns are not working correctly until we add the constraint rules so I will wait with adding this check but I believe you are right.